crates/rune/src/compile/expr_binary.rs

```diff
@@ -83,6 +96,53 @@ fn rhs_needs_of(op: ast::BinOp) -> Needs {
     }
 }
 
+fn compile_conditional_binop(
+   this: &mut Compiler<'_>,
+   lhs: &ast::Expr,
+   rhs: &ast::Expr,
+   bin_op: ast::BinOp,
+   needs: Needs,
+) -> CompileResult<()> {
+    let span = lhs.span().join(rhs.span());
+    
+    let then_label = this.asm.new_label("conditional_then");
+    let end_label = this.asm.new_label("conditional_end");
+
+    this.compile((&*lhs, needs))?;
```

Comment:
What's the expected output for this snippet?

Reply:
A unit `()`, but I believe the current implementation will currently stack error because the `return` doesn't "see" the `true` that needs to be cleaned up.